builtin/providers/aws/resource_aws_lambda_function.go

```diff
@@ -487,3 +541,18 @@ func validateVPCConfig(v interface{}) (map[string]interface{}, error) {
 
 	return config, nil
 }
+
+func validateEnvironment(v interface{}) (map[string]interface{}, error) {
```

Comment:
I am not sure what this one if really for? Are you checking/protecting against having `environments` block specified multiple times?